lib/services.js

```diff
@@ -166,9 +166,12 @@ export default {
         const { objectKey, authInfo, size, contentMD5, metaHeaders,
             contentType, cacheControl, contentDisposition, contentEncoding,
             expires, multipart, headers, overrideMetadata, log,
-            lastModifiedDate } = params;
+            lastModifiedDate, versionId } = params;
         log.trace('storing object in metadata');
         assert.strictEqual(typeof bucketName, 'string');
+
+        const versioningParams = versionId ? { versionId } : {};
```

Comment:
@ttvi for request to metadata when there is no version id, do you prefer:
`/foo/bar` (no query for versionId) or
`/foo/bar?versionId=` empty value for versionId?